test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,509 @@
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+    }
+
+    @After
+    public void tearDown() {
+        ExecutorServiceHelpers.shutdown(executor);
+    }
+
+    /**
+     * This test the getNextStreamCutWithScaleUpTest api with the current streamcut containing one segment0, and written the 10 events,
+     * created the streamCut0, streamCut1, streamCut2, streamCut3, streamCut4 and streamCut5 by calling getNextStreamCut.
+     * Read the events between the streamCut and segment0 has scaled up(segment1, segment2) and written 5 events and validating the numbers of events written
+     * by reading the events from the scaled up segment.
+     */
+    @Test(timeout = 90000)
+    public void getNextStreamCutWithScaleUpTest() throws SegmentTruncatedException, ExecutionException, InterruptedException {
+        String streamName = "testStreamSegment";
+        String streamScope = "testScopeSegment";
+        String readerGroupName = "testReaderGroupSegment";
+        String readerName = UUID.randomUUID().toString();
+
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+        @Cleanup
+        ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(clientConfig);
+        ControllerImpl controller = new ControllerImpl(ControllerImplConfig.builder().clientConfig(clientConfig).build(),
+                connectionFactory.getInternalExecutor());
+
+        StreamConfiguration streamConfiguration = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+        Stream stream = Stream.of(streamScope, streamName);
+
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(streamScope, Utils.buildClientConfig(controllerURI));
+        log.info("Invoking Writer test with Controller URI: {}", controllerURI);
+
+        StreamManager streamManager = StreamManager.create(clientConfig);
+        assertTrue("Creating Scope", streamManager.createScope(streamScope));
+        assertTrue("Creating stream", streamManager.createStream(streamScope, streamName, streamConfiguration));
+
+        @Cleanup
+        BatchClientFactory batchClient = BatchClientFactory.withScope(streamScope, clientConfig);
+
+        List<SegmentRange> ranges = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        List<Segment> list = ranges.stream().map(SegmentRange::getSegment).collect(Collectors.toList());
+        log.info("Segment name ::{}", list.get(0).getScopedName());
+        assertEquals(1, list.size());
```

Comment:
Here, I am ensuring that number of segments should be 1 since, which creating of  StreamConfiguration object, scaling policy has passed as 1.

Reply:
I think instead of creating it manually, we can get headStreamCut here. It will fullfill your both requirement:

- check no of segments
- get initial streamcut.

Reply:
Made the changes.